src/app/content/utils/contentManipulation.ts

```diff
@@ -0,0 +1,28 @@
+import { HTMLAnchorElement } from '@openstax/types/lib.dom';
+import { fromRelativeUrl } from './urlUtils';
+
+const domParser = new DOMParser();
+
+export function addTargetBlankToLinks(htmlString: string): string  {
+  const domNode = domParser.parseFromString(htmlString, 'text/html');
+  domNode.querySelectorAll('a').forEach((a: HTMLAnchorElement) => a.setAttribute('target', '_blank'));
+  return domNode.body.innerHTML;
+}
+
+export const isAbsoluteUrl = (url: string) => {
+  const pattern = /^(http)?s?:\/\//i;
```

Comment:
this matches things it shouldn't you want to move the `s?:` into the parentheses with the `http`

you also don't want to treat urls beginning in a slash, like `/books/asdf/pages/qre` as relative, they aren't fully qualified urls because they don't have the domain or protocol, but they are absolute paths